modules/novatiqIdSystem.js

```diff
@@ -47,18 +47,94 @@ export const novatiqIdSubmodule = {
 
     const configParams = config.params || {};
     const srcId = this.getSrcId(configParams);
+    const sharedId = this.getSharedId(configParams);
+
+    logInfo('NOVATIQ config params: ' + JSON.stringify(configParams));
     logInfo('NOVATIQ Sync request used sourceid param: ' + srcId);
+    logInfo('NOVATIQ Sync request Shared ID: ' + sharedId);
 
     let partnerhost;
     partnerhost = window.location.hostname;
     logInfo('NOVATIQ partner hostname: ' + partnerhost);
 
     const novatiqId = snowflakeId();
-    const url = 'https://spadsync.com/sync?sptoken=' + novatiqId + '&sspid=' + srcId + '&ssphost=' + partnerhost;
+
+    let url = 'https://spadsync.com/sync?sptoken=' + novatiqId + '&sspid=' + srcId + '&ssphost=' + partnerhost;
+
+    // for testing
+    let sharedStatus = 'Not Found';
+
+    // append on the shared ID if we have one
+    if (sharedId != null) {
+      url = url + '&sharedId=' + sharedId;
+      sharedStatus = 'Found';
+    }
+
     ajax(url, undefined, undefined, { method: 'GET', withCredentials: false });
 
     logInfo('NOVATIQ snowflake: ' + novatiqId);
-    return { 'id': novatiqId }
+    return { 'id': novatiqId,
+      'sharedStatus': sharedStatus }
+  },
+
+  useSharedId(configParams) {
+    return typeof configParams.useSharedId != 'undefined' && configParams.useSharedId === true;
+  },
+
+  getFromLocalStorage(cookieOrStorageID) {
+    return window.localStorage.getItem(cookieOrStorageID);
+  },
+
+  getFromSessionStorage(cookieOrStorageID) {
+    return window.sessionStorage.getItem(cookieOrStorageID);
+  },
+
+  // return null if we aren't supposed to use one or we are but there isn't one present
+  getSharedId(configParams) {
+    let sharedId = null;
+    if (this.useSharedId(configParams)) {
+      let cookieOrStorageID = '_pubcid';
+
+      // Has the cookieOrStorageID been redefined?
+      if (typeof configParams.sharedIdName != 'undefined' && configParams.sharedIdName != null && configParams.sharedIdName != '') {
+        cookieOrStorageID = configParams.sharedIdName;
+        logInfo('NOVATIQ sharedID name redefined: ' + cookieOrStorageID);
+      }
+
+      // Check local storage first
+      sharedId = this.getFromLocalStorage(cookieOrStorageID);
+      if (sharedId == null) {
+        // next check session storage
+        sharedId = this.getFromSessionStorage(cookieOrStorageID);
+
+        // OK check cookies
+        if (sharedId == null) {
+          sharedId = this.getCookieValue(cookieOrStorageID);
```

Comment:
is it guaranteed that the sharedid is in the cookie if you get here?

Reply:
no but thats ok, if it's not there the additional parameter isnt added onto the sync request.  I've looked around and gone for the getStorageManager pattern as used in the sharedIdModule. Thanks for pointing out there were existing functions.